src/api-binder/report.ts

```diff
@@ -64,30 +63,46 @@ async function report({ body, opts }: StateReport) {
 	}
 }
 
-async function reportCurrentState(opts: StateReportOpts) {
+interface ReportResult {
+	reported: boolean;
+	state: StateDiff;
+}
```

Comment:
This is created to convey the result of a report attempt. Provides an interface to know what was reported (full and diff) and if it was reported. This easily allows the caller to update last reported states.